包含头文件<math.h>编译报错

代码包含了头文件<math.h>，编译时仍然会报错“对'ceil'未定义的引用”：
```
cc -o output/exec1-21.out -ansi exec1-21.c
/usr/bin/ld: /tmp/ccxM5P5J.o: in function `print_space':
exec1-21.c:(.text+0xf6): undefined reference to `ceil'
collect2: error: ld returned 1 exit status
```

应该在编译命令最后添加`-lm`（必须在源文件名之后）：
```bash
cc -o output/exec1-21.out -ansi exec1-21.c -lm
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -11,5 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
+    - name: Print gcc version
+      run: gcc --version
     - name: Build
       run: make

--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,5 @@
 # C
-output/
+*.o
 *.out
 
 # Visual Studio Code

--- a/ch1/Makefile
+++ b/ch1/Makefile
@@ -1,14 +1,16 @@
 SOURCES = $(shell ls *.c)
-TARGETS = $(addprefix output/,$(SOURCES:.c=.out))
+TARGETS = $(SOURCES:.c=.out)
 CFLAGS = -ansi
 
 all: $(TARGETS)
 
-output/%.out: %.c
-	@[ -d output ] || mkdir output
+%.out: %.o
 	$(CC) -o $@ $(CFLAGS) $<
 
+exec1-21.out: exec1-21.o
+	$(CC) -o $@ $(CFLAGS) $< -lm
+
 clean:
-	rm -r output
+	rm *.o *.out
 
 .PHONY: all clean

